File: vibe-coding/todo-app/frontend_web/src/modules/tasks_mod/application/delete_task/DeleteTaskDto.ts
/** Entrada del caso de uso DeleteTask. */
export class DeleteTaskDto {
    private constructor(public readonly taskId: number) {}

    public static fromPrimitives(primitives: Record<string, unknown>): DeleteTaskDto {
        return new DeleteTaskDto(Number(primitives["taskId"] ?? 0));
    }
}

File: vibe-coding/todo-app/backend_web/src/modules/tasks_mod/application/delete_task/delete_task_dto.py
from dataclasses import dataclass
from typing import Any, Self, final

from src.modules.tasks_mod.domain.enums.task_field_enum import TaskFieldEnum


@final
@dataclass(frozen=True, slots=True)
class DeleteTaskDto:
    """Entrada del caso de uso DeleteTask."""

    task_id: int

    @classmethod
    def from_primitives(cls, primitives: dict[str, Any]) -> Self:
        return cls(task_id=int(primitives.get(TaskFieldEnum.TASK_ID, 0) or 0))
